From 8c33b6d46b2874890813237b1c67c42c6d14b19f Mon Sep 17 00:00:00 2001
From: Jeegar Patel <jeegar.patel@xilinx.com>
Date: Tue, 22 Sep 2020 05:05:28 -0700
Subject: [PATCH 1625/1852] drm: xlnx: bridge: Fix NULL_RETURNS type warning

commit d9230d0febbaea33865a914ee144760dc6bc8d30 from
https://github.com/Xilinx/linux-xlnx.git

This patch fix NULL_RETURNS type coverity warning.

Assigning: "cmd" = "NULL" return value from "kzalloc".

Dereferencing a pointer that might be "NULL" "cmd" when calling
"strncpy_from_user".

Signed-off-by: Jeegar Patel <jeegar.patel@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/gpu/drm/xlnx/xlnx_bridge.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/xlnx/xlnx_bridge.c b/drivers/gpu/drm/xlnx/xlnx_bridge.c
index 016cfcf6fb07..76456f527d00 100644
--- a/drivers/gpu/drm/xlnx/xlnx_bridge.c
+++ b/drivers/gpu/drm/xlnx/xlnx_bridge.c
@@ -347,6 +347,9 @@ static ssize_t xlnx_bridge_debugfs_write(struct file *f, const char __user *buf,
 		int ret;
 
 		cmd = kzalloc(size, GFP_KERNEL);
+		if (!cmd)
+			return -ENOMEM;
+
 		ret = strncpy_from_user(cmd, buf, size);
 		if (ret < 0) {
 			pr_err("%s %d failed to copy the command  %s\n",
-- 
2.31.1

